### PR TITLE
Added documentation for withastro/astro#4947

### PR DESCRIPTION
**Pull request by JuanM04**
#### What kind of changes does this PR include?

- New or updated content

#### Description

Adds documentation for withastro/astro#4947

### CODE REVIEW

File: src/pages/en/reference/integrations-reference.md [hunk 1]
```diff
@@ -154,6 +162,21 @@ The **`stage`** denotes how this script (the `content`) should be inserted. Some
     injectScript('page-ssr', 'import "global-styles.css";');
     ```
 
+#### `injectWatchTarget` option
+
+**Type:** `InjectedWatchTarget`
+
+If your integration depends on some configuration file that Vite doesn't watch and/or need a full dev server restart to take effect, add it with `injectWatchTarget`. Whenever that file changes, the integration will be reloaded (you can check when a reload happens with `isConfigReload`).
```
Comment (delucis) [inline src/pages/en/reference/integrations-reference.md:169]: Is it _just_ the integration that gets reloaded or is it the whole dev server?

```suggestion
If your integration depends on some configuration file that Vite doesn't watch and/or need a full dev server restart to take effect, add it with `injectWatchTarget`. Whenever that file changes, the Astro dev server will be reloaded (you can check when a reload happens with `isConfigReload`).
```

**Review by delucis — COMMENTED** [review]
Thanks @JuanM04! Looks good. I’m going to mark this PR as a draft until https://github.com/withastro/astro#4947 is merged, to make sure we don’t merge it ahead of time 🙌

File: src/pages/en/reference/integrations-reference.md [hunk 2]
```diff
@@ -154,6 +162,21 @@ The **`stage`** denotes how this script (the `content`) should be inserted. Some
     injectScript('page-ssr', 'import "global-styles.css";');
     ```
 
+#### `injectWatchTarget` option
+
+**Type:** `InjectedWatchTarget`
+
+If your integration depends on some configuration file that Vite doesn't watch and/or need a full dev server restart to take effect, add it with `injectWatchTarget`. Whenever that file changes, the Astro dev server will be reloaded (you can check when a reload happens with `isConfigReload`).
```
Comment (sarah11918) [inline src/pages/en/reference/integrations-reference.md:169]:
```suggestion
If your integration depends on some configuration file that Vite doesn't watch and/or needs a full dev server restart to take effect, add it with `injectWatchTarget`. Whenever that file changes, the Astro dev server will be reloaded (you can check when a reload happens with `isConfigReload`).
```
I think the verb tense "needs" is required here.

**Comment by sarah11918** [comment]
Thanks, @JuanM04, looks great!  🥳  

@delucis - Procedural question: how do we know when/whether this is to be labeled as a draft vs `merge-on-release`?  😅

**Comment by delucis** [comment]
> Procedural question: how do we know when/whether this is to be labeled as a draft vs merge-on-release?

I guess draft until https://github.com/withastro/astro/pull/4947 is accepted and then `merge-on-release` if the upstream PR won’t be merged until a minor release?

**Pull request merged**